### PR TITLE
Include ai.operation.parentId from Activity.ParentSpanId instead of Activity.Parent.SpanId

### LINKED ISSUE #20433: [BUG] TelemetryItem tag ai.operation.parentId not set when Activity.Parent is null when Activity.ParentSpanId is defined
**Describe the bug**
When starting a new Activity with passing a parent ActivityContext that has ParentSpanId set the Parent property on Activity is null while the ParentSpanId is set.
In the current implementation the Activity.ParentSpanId is not used and that results in the TelemetryItem not having the tag ai.operation.parentId set.

Examples when this happens is if one or more activities are not part of the ActivitySource that is being monitored. Then the started activities has the parent set manually from ActivityContext to maintain the link between them regarding parents.

**Expected behavior**
The ai.operation.parentId tag should be set when Activity.ParentSpanId is set

**Actual behavior (include Exception or Stack Trace)**
The ai.operation.parentId is not set when the Activity is created with a ActivityContext passed as the Parent property is null

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Create a ActivitySource
2. Start a Activity from the ActivitySource
3. Start another Activity from the ActivitySource but this time pass a ActivityContext with the TraceId and SpanId set to the previous Activity's TraceId and SpanId
4. When transmitted the 2nd Activity produces a TelemetryItem with ai.operation.id set but without ai.operation.parentId set.

**Environment:**
 - Name and version of the Library package used: Azure.Monitor.OpenTelemetry.Exporter 1.0.0-beta.2

### PR DESCRIPTION
**Pull request by mangeg**
Fixes #20433

When starting a new Activity with passing parent context the Parent property of the Activity is null. With the current implementation that means the ParentSpanId is not set on the TelemetryItem.

To maintain the previous functionally the Activity could be created with a ParentContext with no ParentSpanId set.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by vishweshbankwar** [comment]
Hi @mangeg - Could you please look into conflicts?
I created the same PR since I did not know about this one.
https://github.com/Azure/azure-sdk-for-net/pull/22061

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TagsTests.cs [hunk 1]
```diff
@@ -326,7 +326,7 @@ private static Activity CreateTestActivity(IEnumerable<KeyValuePair<string, obje
             var eventTimestamp = DateTime.UtcNow;
             var traceId = ActivityTraceId.CreateRandom();
 
-            var parentSpanId = ActivitySpanId.CreateRandom();
+            var parentSpanId = default(ActivitySpanId);
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TagsTests.cs:329]: @mangeg - I am not sure why this change is included.

Comment (mangeg) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TagsTests.cs:329]: No, that was not supposed to be there. Must have been left in from testing.
Will make a update to revert that change.

**Review by cijothomas — APPROVED** [review]
LGTM.

We may want to add an unit test for the scenario where the parent is in-proc as well, if such a test doesn't exist.

**Comment by vishweshbankwar** [comment]
> LGTM.
> 
> We may want to add an unit test for the scenario where the parent is in-proc as well, if such a test doesn't exist.

Good Point - I will get this added in one of the other PRs.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TagsTests.cs (hunk 1 above)
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TagsTests.cs]: Thanks!
